### PR TITLE
thrift_proxy: fix crash on invalid transport/protocol

### PR DESCRIPTION
**Pull request by zuercher**
Transport/protocol decoder errors that occur before the connection manager
initializes an ActiveRPC to track the request caused a crash. Modifies the
connection manager to handle this case, terminating the downstream
connection.

Risk Level: low
Testing: test case that triggers crash
Docs Changes: n/a
Release Notes: added

Signed-off-by: Stephan Zuercher <zuercher@gmail.com>

### CODE REVIEW

**Comment by zuercher** [comment]
@brian-pane do you have time to look at this?

File: test/extensions/filters/network/thrift_proxy/conn_manager_test.cc [hunk 1]
```diff
@@ -509,6 +509,16 @@ TEST_F(ThriftConnectionManagerTest, OnDataHandlesTransportApplicationException)
   EXPECT_EQ(0U, stats_.request_active_.value());
 }
 
+TEST_F(ThriftConnectionManagerTest, OnDataHandlesGarbageRequest) {
```
Comment (htuch) [inline test/extensions/filters/network/thrift_proxy/conn_manager_test.cc]: Can you add `//` comments above each `TEST_F` explaining that it's a regression for the specific bug.

File: source/extensions/filters/network/thrift_proxy/conn_manager.cc [hunk 2]
```diff
@@ -77,8 +77,14 @@ void ConnectionManager::dispatch() {
   } catch (const EnvoyException& ex) {
     ENVOY_CONN_LOG(error, "thrift error: {}", read_callbacks_->connection(), ex.what());
 
-    // Use the current rpc to send an error downstream, if possible.
-    rpcs_.front()->onError(ex.what());
+    if (!rpcs_.empty()) {
```
Comment (htuch) [inline source/extensions/filters/network/thrift_proxy/conn_manager.cc]: Nit: can you invert the logic here to make it easier to read?

**Pull request merged**